sonic-chassisd/scripts/chassisd

```diff
@@ -746,26 +1228,84 @@ class ChassisdDaemon(daemon_base.DaemonBase):
         else:
             self.log_warning("Caught unhandled signal '{}' - ignoring...".format(SIGNALS_TO_NAMES_DICT[sig]))
 
+    def submit_dpu_callback(self, module_index, admin_state):
+        try_get(self.module_updater.chassis.get_module(module_index).set_admin_state, admin_state, default=False)
+        pass
+
+    def set_initial_dpu_admin_state(self):
+        """Send admin_state trigger once to modules those are powered up"""
+        threads = []
+        for module_index in range(0, self.module_updater.num_modules):
+            op = None
+            # Get midplane state of DPU
+            module_info_dict = self.module_updater._get_module_info(module_index)
+            module_name = self.platform_chassis.get_module(module_index).get_name()
+            state_db = daemon_base.db_connect("STATE_DB")
+            midplane_table = swsscommon.Table(state_db, CHASSIS_MIDPLANE_INFO_TABLE)
+            fvs = midplane_table.get(module_name)
+            midplane_state = 'False'
+            if isinstance(fvs, list) and fvs[0] is True:
+                fvs = dict(fvs[-1])
+                midplane_state = fvs[CHASSIS_MIDPLANE_INFO_ACCESS_FIELD]
+
+            try:
+                if module_info_dict is not None:
+                    # Get admin state of DPU
+                    key = module_info_dict[CHASSIS_MODULE_INFO_NAME_FIELD]
+                    admin_state = self.module_updater.get_module_admin_status(key)
+                    if admin_state == 'up' and midplane_state == 'False':
+                        # startup DPU
+                        op = MODULE_ADMIN_UP
+
+                    elif admin_state != 'up' and midplane_state == 'True':
+                        # shutdown DPU
+                        op = MODULE_ADMIN_DOWN
+
+                    # Initialize DPU_STATE DB table on bootup
+                    dpu_state_key = "DPU_STATE|" + module_name
+                    self.module_updater.update_dpu_state(dpu_state_key, admin_state.upper())
```

Comment:
Why do we update the midplane state with the configured `admin_state` instead of checking the reachability of the midplane here?

Reply:
We need to query the operation state of the interface connected to the DPU. And if the oper state is up, the we should test the reachability of IP address of the interface on the DPU side

Reply:
During the initialization of the system we want to make sure the system comes up correctly with the configured state, which is the admin_state.  During boot up the operational state of the DPUs could be anything. So, 
1. Get the admin_state and initialize the DPUs to that state
2. Set the DPU_STATE table as per the admins state
3. Soon the periodic update of the DPU_STATE will kick which will check for the operational state of the DPU and update it

Reply:
Assuming before we call the `set_initial_dpu_admin_state` the DPUs are powered off but the CONFIG_DB entry is set to `admin_state=up` we are setting the `dpu_midplane_link_state=up` in the CHASSIS_STATE_DB before we even power on the DPUs (which is done in the `submit_dpu_callback` later in this function), if the DPU power on fails or if we do not achieve midplane reachability we have the wrong data in the CHASSIS_STATE_DB